dev/tools/controllerbuilder/scaffold/controller.go

```diff
@@ -34,29 +34,78 @@ const (
 	directControllerRelPath = "pkg/controller/direct"
 )
 
-func Scaffold(path string, cArgs *ccTemplate.ControllerArgs) error {
+func Scaffold(service, kind string, cArgs *ccTemplate.ControllerArgs) error {
+	var errs []error
+	if err := generateController(service, kind, cArgs); err != nil {
+		errs = append(errs, err)
+	}
+	if err := generateControllerHelpers(service, kind, cArgs); err != nil {
+		errs = append(errs, err)
+	}
+	if len(errs) != 0 {
+		var finalError []string
```

Comment:
Nit: use `return errors.Join(errs...)`.  I think it even checks if len(errs) == 0 for you :-)